akka-bbb-apps/src/main/scala/org/bigbluebutton/LockSettingsUtil.scala

```diff
@@ -0,0 +1,75 @@
+package org.bigbluebutton
+
+import org.bigbluebutton.common2.msgs.{ BbbCommonEnvCoreMsg, BbbCoreEnvelope, BbbCoreHeaderWithMeetingId, MessageTypes, MuteUserInVoiceConfSysMsg, MuteUserInVoiceConfSysMsgBody, Routing }
+import org.bigbluebutton.core.models.{ Roles, Users2x, VoiceUserState, VoiceUsers }
+import org.bigbluebutton.core.running.{ LiveMeeting, OutMsgRouter }
+import org.bigbluebutton.core2.{ MeetingStatus2x }
+
+object LockSettingsUtil {
+
+  def muteUserInVoiceConf(liveMeeting: LiveMeeting, outGW: OutMsgRouter, vu: VoiceUserState, mute: Boolean): Unit = {
+    val routing = Routing.addMsgToClientRouting(MessageTypes.BROADCAST_TO_MEETING, liveMeeting.props.meetingProp.intId, vu.intId)
+    val envelope = BbbCoreEnvelope(MuteUserInVoiceConfSysMsg.NAME, routing)
+    val header = BbbCoreHeaderWithMeetingId(MuteUserInVoiceConfSysMsg.NAME, liveMeeting.props.meetingProp.intId)
+
+    val body = MuteUserInVoiceConfSysMsgBody(liveMeeting.props.voiceProp.voiceConf, vu.voiceUserId, mute)
+    val event = MuteUserInVoiceConfSysMsg(header, body)
+    val msgEvent = BbbCommonEnvCoreMsg(envelope, event)
+
+    outGW.send(msgEvent)
+  }
+
+  def applyMutingOfUsers(mute: Boolean, liveMeeting: LiveMeeting, outGW: OutMsgRouter): Unit = {
+    VoiceUsers.findAll(liveMeeting.voiceUsers) foreach { vu =>
+      Users2x.findWithIntId(liveMeeting.users2x, vu.intId).foreach { user =>
+        if (user.role == Roles.VIEWER_ROLE) {
```

Comment:
Will this catch voice users that don't have an associated User? (duplicate voice users and dial-in)